packages/router/src/grpc/view-protocol-server/balances.ts

```diff
@@ -109,6 +120,39 @@ class BalancesAggregator {
     );
     currentAmount.lo = newAmount.lo;
     currentAmount.hi = newAmount.hi;
+
+    await this.aggregateEquivalentValues(valueView, toAdd);
+  }
+
+  private async aggregateEquivalentValues(valueView: ValueView, toAdd: SpendableNoteRecord) {
+    const assetId = getAssetIdFromRecord.optional()(toAdd);
+    if (!assetId?.inner) return;
+
+    const amount = getAmount(valueView);
+
+    const equivalentValues: EquivalentValue[] = [];
+
+    for (const price of this.estimatedPriceByPricedAsset[uint8ArrayToBase64(assetId.inner)] ?? []) {
```

Comment:
Not strictly necessary, but may be a nicer api to have a small external class to manage this object so you could do something like:
```
this.estimatedPriceByPricedAsset.add(assetId)
```
and the class would handle the object keys and getting as well.

It may help also moving the initialization logic out as well to this helper class.

Reply:
Lemme do this as a quick-follow up PR in case there are disagreements about how it's implemented.